connector/connector.go

```diff
@@ -97,3 +97,8 @@ type RefreshConnector interface {
 	// changes since the token was last refreshed.
 	Refresh(ctx context.Context, s Scopes, identity Identity) (Identity, error)
 }
+
+// UserInfoConnector represents connectors that support the user info endpoint
```

Comment:
Hmm that's interesting. We're relaying the connector's userinfo data! I would have assumed we'd have a single userinfo endpoint in Dex, which merely decodes the ID token info, or renders the identity as far as Dex is concerned 🤔

Reply:
OK, I've looked at the `/userinfo` handler code now -- this allows connectors to _enrich_ the user info response, right?

Reply:
That's my understanding of this and I think that's probably the right thing to do, I was reading through [the standard claims](https://connect2id.com/products/server/docs/api/userinfo#claims) for the userinfo endpoint and there's a whole lot of info in there that Dex wouldn't normally have but upstream providers may provide somehow